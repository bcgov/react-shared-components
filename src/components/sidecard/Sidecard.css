.heading-style {
  font-weight: 600;
  font-style: normal;
  color: #ffffff;
  text-align: left;
}

.bluegrey-heading {
  padding-top: 10px;
  padding-left: 15px;
  padding-bottom: 9px;
}

.bluegrey-heading h2 {
  margin-bottom: 0px;
}

.bluegrey-content {
  height: auto;
  background: inherit;
  background-color: #f2f2f2;
  box-sizing: border-box;
  padding-top: 15px;
}

.content-style {
  padding-right: 15px;
  padding-bottom: 10px;
  padding-left: 15px;
  display: flex;
  flex-direction: column;
}

.container-background {
  background-color: rgba(56, 89, 138, 1);
  font-weight: 400;
  font-style: normal;
  font-size: 12px;
  text-align: left;
}

.submitted-header h2 {
  color: #fff !important;
}

.submitted-content {
  background-color: #eee;
  padding: 30px;
  margin-bottom: 20px;
}

.submit-container {
  background-color: #38598a;
  border-bottom: solid 4px #fc0;
  color: #fff;
  margin-bottom: 20px;
  padding: 30px;
}

.bluegrey-container {
  margin-bottom: 20px;
}

.submit-container h2 {
  color: #fff;
}

.dashboard-spacing {
  padding-left: 25px;
  max-width: 400px;
}

.wide-dashboard-spacing {
  padding-left: 25px;
  max-width: 458px;
}

img {
  max-width: 100%;
}

@media (max-width: 575.98px) {
  .dashboard-spacing {
    padding-left: 10px;
    padding-right: 10px;
  }

  .wide-dashboard-spacing {
    padding-left: 0;
  }
}

.side-card-row {
  white-space: nowrap;
  display: flex;
}

.side-card-icon {
  width: 35px;
  height: 35px;
  padding-bottom: 10px;
  margin-bottom: 5px;
}

.round-icon-wrapper {
  padding-top: 8px;
  margin-right: 10px;
}

.side-card-title {
  align-self: center;
  font-size: 20px;
  font-weight: Semi-bold;
  width: 100%;
  white-space: normal;
  padding-top: 10px;
}

.large-title {
  font-size: 20px;
  font-weight: Semi-bold;
  width: 80%;
  white-space: normal;
  padding-top: 0 !important;
}

@media (max-width: 950px) {
  .side-card-title {
    padding-top: 0px !important;
  }
}

@media (max-width: 865px) {
  .wide-dashboard-spacing {
    padding-left: 0px;
  }
}
