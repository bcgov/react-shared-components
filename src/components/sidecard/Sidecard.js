/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";
import PropTypes from "prop-types";

import "./Sidecard.css";

export const Sidecard = ({
  sideCard: { heading, content, type, image, imageLink, isWide, icon },
}) => {
  let sideCardCss = "dashboard-spacing";
  if (isWide) sideCardCss = "wide-dashboard-spacing";

  const sideCardTitleCss =
    heading.length > 30 ? "large-title" : "side-card-title";

  return (
    <div className={sideCardCss} style={{ position: "relative" }}>
      <div className="row">
        <div className="col-12 pt-2" style={{ position: "relative" }}>
          <p />
          {type === "grey" && (
            <section
              id="grey-section"
              className="submit-container"
              style={{
                backgroundColor: "#F2F2F2",
                color: "#000",
                border: "none",
              }}
            >
              <h2 style={{ color: "#000" }}>{heading}</h2>
              <div className="submit-content">
                <span>{content}</span>
              </div>
            </section>
          )}
          {type === "blue" && (
            <section id="blue-section" className="submit-container">
              <h2 className="heading-style">{heading}</h2>
              <div className="submit-content">
                <span>{content}</span>
                {image && imageLink && (
                  <a href={imageLink} target="_blank" rel="noopener noreferrer">
                    <img
                      src={image}
                      alt="imagelink"
                      height="65px"
                      width="310px"
                    />
                  </a>
                )}
              </div>
            </section>
          )}
          {type === "bluegrey" && (
            <section id="bluegrey-section" className="bluegrey-container">
              <div className="container-background bluegrey-heading">
                {!icon && <h2 className="heading-style">{heading}</h2>}
                {icon && (
                  <h2 className="heading-style">
                    <div className="row">
                      <div className="side-card-row">
                        <div className="round-icon-wrapper">{icon}</div>
                        <div className={sideCardTitleCss}>{heading}</div>
                      </div>
                    </div>
                  </h2>
                )}
              </div>
              <div className="bluegrey-content">
                <span className="content-style">{content}</span>
              </div>
            </section>
          )}
        </div>
      </div>
    </div>
  );
};

Sidecard.propTypes = {
  sideCard: PropTypes.shape({
    heading: PropTypes.string.isRequired,
    content: PropTypes.array.isRequired,
    type: PropTypes.string.isRequired,
    image: PropTypes.string,
    imageLink: PropTypes.string,
    isWide: PropTypes.bool,
    icon: PropTypes.element,
  }).isRequired,
};
