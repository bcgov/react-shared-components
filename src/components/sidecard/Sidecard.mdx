import { Meta, Story } from "@storybook/addon-docs";
import { Sidecard } from "./Sidecard";

<Meta title="Sidecard" component={Sidecard} />

# Sidecard

Sidecards are an excerpt of text that has been pulled out and used as a visual clue
to draw the eye to the text. They are used to help direct a user's attention
to important pieces of information.

## When to Use

Differentiating a block of text from the content that surrounds it, for example:

- Important snippets of information
- Quotes
- Examples
- Additional information about the page

## When not to Use

- Warning notifications
- Page alerts

## API

| Prop     | Type   | Required | Description                                | Default |
| -------- | ------ | -------- | ------------------------------------------ | ------- |
| sidecard | object | true     | object with details regarding the sidecard |         |

The sidecard object has its own properties that need to be present within it, as shown below:

| Prop      | Type    | Required | Description                                                                    | Default |
| --------- | ------- | -------- | ------------------------------------------------------------------------------ | ------- |
| heading   | string  | true     | text representing the label you wish to display as the heading of the sidecard |         |
| content   | array   | true     | list of strings for content to be present in the sidecard                      |         |
| type      | string  | true     | represents the type of sidecard you wish to render                             |         |
| image     | string  | false    | url to your image to display                                                   |         |
| imageLink | string  | false    | url that your image links to on click                                          |         |
| isWide    | boolean | false    | indicates if your sidecard is extra wide                                       | false   |
| icon      | element | false    | html (jsx) representing the icon you wish to display                           |         |

<br />

## Information (Blue) Sidecard

Used to provide useful additional information to compliment content present on the screen.

<Story name="Information (Blue) Sidecard" id="sidecard--blue"></Story>

```bash
const sideCard = {
  heading: "Heading",
  content: [
    "This is where I put my content for the sidecard"
  ],
  type: "blue"
};

<Sidecard sideCard={sideCard} />;
```

## Notice (Grey) Sidecard

Used to typically provide info about a collection notice, or similar information.

<Story name="Notice (Grey) Sidecard" id="sidecard--grey"></Story>

```bash
const sideCard = {{ ...sideCard, type: "grey" }};

<Sidecard sideCard={sideCard} />;
```

## Information (Blue grey) Sidecard

Used to provide useful additional information to compliment content present on the screen.

<Story name="Information (Blue grey) Sidecard" id="sidecard--blue-grey"></Story>

```bash
const sideCard = {{ ...sideCard, type: "bluegrey" }};

<Sidecard sideCard={sideCard} />;
```

## Sidecard with Icon

Can be used with any of the above types of sidecards, to add an icon to the heading.

<Story name="Sidecard with Icon" id="sidecard--with-icon"></Story>

```bash
const sideCard = {{ ...sideCard, icon: <FaIdCard className="bcgov-side-card-icon" /> }}

<Sidecard sideCard={sideCard} />;
```
