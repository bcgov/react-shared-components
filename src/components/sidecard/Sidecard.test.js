import React from "react";
import testBasicSnapshot from "../../TestHelper";
import { Sidecard } from "./Sidecard";

describe("Sidecard Component", () => {
  const sideCard = {
    heading: "header",
    content: ["content"],
    type: "grey",
  };

  test("Matches the grey card snapshot", () => {
    testBasicSnapshot(<Sidecard sideCard={sideCard} />);
  });

  test("Matches the blue card snapshot without image", () => {
    testBasicSnapshot(<Sidecard sideCard={{ ...sideCard, type: "blue" }} />);
  });

  test("Matches the blue card snapshot with image", () => {
    testBasicSnapshot(
      <Sidecard
        sideCard={{
          ...sideCard,
          type: "blue",
          image: "image",
          imageLink: "imageLink",
        }}
      />
    );
  });

  test("Matches the bluegrey card snapshot", () => {
    testBasicSnapshot(
      <Sidecard sideCard={{ ...sideCard, type: "bluegrey" }} />
    );
  });

  test("Matches the snapshot when it is wide", () => {
    testBasicSnapshot(<Sidecard sideCard={{ ...sideCard, isWide: true }} />);
  });

  test("Matches the snapshot when heading is long", () => {
    testBasicSnapshot(
      <Sidecard
        sideCard={{
          ...sideCard,
          heading:
            "heading is quite long so we can use custom styling to see it render",
        }}
      />
    );
  });
});
