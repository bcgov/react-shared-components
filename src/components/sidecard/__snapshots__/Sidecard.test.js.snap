// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidecard Component Matches the blue card snapshot with image 1`] = `
<div
  className="bcgov-dashboard-spacing"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="col-12 pt-2"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <p />
    <section
      className="bcgov-submit-container"
      id="blueimagecard"
    >
      <h2
        className="bcgov-heading-style"
      >
        header
      </h2>
      <div
        className="bcgov-submit-content"
      >
        <span>
          content
        </span>
        <a
          href="imageLink"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="imagelink"
            height="65px"
            src="image"
            width="310px"
          />
        </a>
      </div>
    </section>
  </div>
</div>
`;

exports[`Sidecard Component Matches the blue card snapshot without image 1`] = `
<div
  className="bcgov-dashboard-spacing"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="col-12 pt-2"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <p />
    <section
      className="bcgov-submit-container"
      id="bluecard"
    >
      <h2
        className="bcgov-heading-style"
      >
        header
      </h2>
      <div
        className="bcgov-submit-content"
      >
        <span>
          content
        </span>
      </div>
    </section>
  </div>
</div>
`;

exports[`Sidecard Component Matches the bluegrey card snapshot 1`] = `
<div
  className="bcgov-dashboard-spacing"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="col-12 pt-2"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <p />
    <section
      className="bcgov-bluegrey-container"
      id="bluegreycard"
    >
      <div
        className="bcgov-container-background bcgov-bluegrey-heading"
      >
        <h2
          className="bcgov-heading-style"
        >
          header
        </h2>
      </div>
      <div
        className="bcgov-bluegrey-content"
      >
        <span
          className="bcgov-content-style"
        >
          content
        </span>
      </div>
    </section>
  </div>
</div>
`;

exports[`Sidecard Component Matches the grey card snapshot 1`] = `
<div
  className="bcgov-dashboard-spacing"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="col-12 pt-2"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <p />
    <section
      className="bcgov-submit-container"
      id="greycard"
      style={
        Object {
          "backgroundColor": "#F2F2F2",
          "border": "none",
          "color": "#000",
        }
      }
    >
      <h2
        style={
          Object {
            "color": "#000",
          }
        }
      >
        header
      </h2>
      <div
        className="bcgov-submit-content"
      >
        <span>
          content
        </span>
      </div>
    </section>
  </div>
</div>
`;

exports[`Sidecard Component Matches the snapshot when heading is long 1`] = `
<div
  className="bcgov-dashboard-spacing"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="col-12 pt-2"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <p />
    <section
      className="bcgov-submit-container"
      id="longheadingcard"
      style={
        Object {
          "backgroundColor": "#F2F2F2",
          "border": "none",
          "color": "#000",
        }
      }
    >
      <h2
        style={
          Object {
            "color": "#000",
          }
        }
      >
        heading is quite long so we can use custom styling to see it render
      </h2>
      <div
        className="bcgov-submit-content"
      >
        <span>
          content
        </span>
      </div>
    </section>
  </div>
</div>
`;

exports[`Sidecard Component Matches the snapshot when it is wide 1`] = `
<div
  className="bcgov-wide-dashboard-spacing"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="col-12 pt-2"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <p />
    <section
      className="bcgov-submit-container"
      id="widecard"
      style={
        Object {
          "backgroundColor": "#F2F2F2",
          "border": "none",
          "color": "#000",
        }
      }
    >
      <h2
        style={
          Object {
            "color": "#000",
          }
        }
      >
        header
      </h2>
      <div
        className="bcgov-submit-content"
      >
        <span>
          content
        </span>
      </div>
    </section>
  </div>
</div>
`;
